addons/web/static/tests/helpers/test_env.js

```diff
@@ -23,15 +23,15 @@ odoo.define('web.test_env', async function (require) {
             qweb = new owl.QWeb({ templates: session.owlTemplates });
         }
         const database = {
-            parameters: {
+            parameters: Object.assign({
                 code: "en_US",
                 date_format: '%m/%d/%Y',
                 decimal_point: ".",
                 direction: 'ltr',
                 grouping: [],
                 thousands_sep: ",",
                 time_format: '%H:%M:%S',
-            },
+            }, (env && env.database && env.database.parameters) ? env.database.parameters : {}),
         };
         const defaultEnv = {
             _t: env._t || Object.assign((s => s), { database }),
```

Comment:
I think all of this should somewhat look like this:
```js
const _t = s => s;
Object.assign(_t, {
   database: Object.assign({
      parameters: Object.assign({
         code: "en_US",
         // ...
      }, env._t && env._t.database && env._t.database.parameters ? env._t.database.parameters : {}),
   }, env._t && env._t.database ? env._t.database : {}),
}, env._t ? env._t : {})
```

It's verbose and not great to read without object spread operator (`{...}`), chain operator (`?.`) and nullish coalescing operator `??`, but we can't use them for the moment (object spread operator could, but we don't have green light of JS team)

Reply:
With little effort, you can make this look like some [brainfuck](https://en.wikipedia.org/wiki/Brainfuck) code.

Reply:
Since we don't have the RTL test, this is not usefull anymore.